### PR TITLE
Feature/tao 3218 metadata csv to ontology import

### PR DESCRIPTION
**Pull request by siwane**

### CODE REVIEW

**Comment by krampstudio** [comment]
Could you please add someone from the backend team in the review loop.

File: models/classes/metadata/metadataImporter.conf.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php
```
Comment (bugalot) [inline models/classes/metadata/metadataImporter.conf.php]: Having such a configuration file in models/classes/ seems not acceptable

File: models/classes/metadata/injector/OntologyMetadataInjector.php [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+            try {
+                $data[$name] = $reader->getValue($dataSource);
+            } catch (MetadataReaderNotFoundException $e) {
+                $errors[$name] = $e->getMessage();
+            }
+        }
+
+        if (! empty($errors)) {
+            foreach ($errors as $name => $error) {
+                \common_Logger::i('Error on injector "' . __CLASS__ . '" with reader "' . $name . '" : ' . $error);
```
Comment (bugalot) [inline models/classes/metadata/injector/OntologyMetadataInjector.php]: Could you please change to debug level?

File: models/classes/metadata/injector/OntologyMetadataInjector.php [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+            try {
+                $writer->write($resource, $data, $dryrun);
+            } catch (MetadataWriterException $e) {
+                $errors[$name] = $e->getMessage();
+            }
+        }
+
+        if (! empty($errors)) {
+            foreach ($errors as $name => $error) {
+                \common_Logger::i('Error on injector "' . __CLASS__ . '" with writer "' . $name . '" : ' . $error);
```
Comment (bugalot) [inline models/classes/metadata/injector/OntologyMetadataInjector.php]: Could you please log level to debug ?

File: models/classes/metadata/writer/ontologyWriter/PropertyWriter.php [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            $validators = ValidationRuleRegistry::getRegistry()->getValidators($this->getPropertyToWrite());
+        } catch (\common_exception_NotFound $e) {
+            throw new MetadataWriterException($e->getMessage());
+        }
+
+        $validated = true;
+        foreach ($validators as $validator) {
+            if (! $validator->evaluate($data)) {
+                $validated = false;
+                \common_Logger::i('Unable to validate value for property "' . $this->getPropertyToWrite()->getUri() . '"' .
```
Comment (bugalot) [inline models/classes/metadata/writer/ontologyWriter/PropertyWriter.php]: Please change log level to debug.

File: models/classes/metadata/writer/ontologyWriter/PropertyWriter.php [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        $propertyValue = $this->format($data);
+        if ($this->validate($propertyValue)) {
+            if (! $dryrun) {
+                if (! $resource->setPropertyValue($this->getPropertyToWrite(), $propertyValue)) {
+                    throw new MetadataWriterException(
+                        'A problem has occurred during writing property "' . $this->getPropertyToWrite()->getUri() . '".'
+                    );
+                }
+            }
+            \common_Logger::i('Valid property "'. $this->getPropertyToWrite()->getUri() .'" ' .
```
Comment (bugalot) [inline models/classes/metadata/writer/ontologyWriter/PropertyWriter.php]: Please change log level to debug

File: models/classes/metadata/import/OntologyMetadataImporter.php [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+        // Global report
+        $report = \common_report_Report::createInfo('Report of metadata import.');
+
+        // Foreach line of dateSource
+        foreach ($data as $uri => $dataSource) {
+            try {
+
+                // Check if resource exists
+                var_dump($uri);
```
Comment (bugalot) [inline models/classes/metadata/import/OntologyMetadataImporter.php]: Woot!

**Pull request merged**